Lesson 7 tasks on Complex Components

lera-kolesnik opened the pull request:
with descriptions inside the commits

Commit:
Added task Учебный проект: Лента-Меню

lera-kolesnik committed:
Completed first task in Lesson 7

(7-1) Implemented RibbonMenu class for displaying selected category items

Commit:
Added task Учебный проект: Модальное окно

lera-kolesnik committed:
Completed second task in Lesson 7

(7-2) Create Modal class for universal modal window functionality
Using methods for setting the title, body, and closing the modal window, can be opened from external code and supports closing by clicking the [X] button in modal__close or pressing the esc key

lera-kolesnik committed:
Merge branch 'master' into master

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

lera-kolesnik committed:
Completed third task in Lesson 7

(7-3) Implement StepSlider component  with click functionality: slider changes its value when clicked

jsru commented:
Добавляю преподавателя (@koalex) для код-ревью.
